### ISSUE TITLE
Injected Bean inside ViewModel in UnitTests override stubbing declareMock()

### ISSUE THREAD

**Issue by LukasStancikas** [InsertKoinIO/koin#643]
I am writing UnitTests for ViewModel/StateMachine (using MVI pattern). The ViewModel has API bean injected inside (built with Retrofit)

```
class UserViewModel(
    private val API: API
)

object UserModule {
    fun get(): Module {
        return module {
            viewModel { UserViewModel(api = get()) }
        }
    }
}

object NetworkModule {
    fun get(): Module {
        return module {
            ... // various retrofit Beans
            single{ Api(retrofit = get()) }
        }
    }
}
```
This is how I try to stub the calls in UnitTests, but **it still uses the old implementation and does network calls**

```
class UserTest : KoinTest {
    private val viewModel: UserViewModel by inject()

    @Before
    fun before() {
        startKoin {
            modules(
                listOf(
                    UserModule.get(),
                    NetworkModule.get()
                )
            )
        }
    }

    @After
    fun after() {
        stopKoin()
    }
}
```
**Tests**
```
@Test
fun `user test passing`() {
    val mockUserData = UserData("name", "surname")

    declareMock<Api> {
        given(this.getUserData()).willReturn { Single.just(mockUserData) }
    }

    viewModel.refresh() // calls api.getUserData() inside

    ... expecting mockUserData but getting real api call
}

@Test
fun `user test failing`() {
    declareMock<Api> {
        given(this.getUserData()).willReturn { Single.error(Exception("mock")) }
    }

    viewModel.refresh() // calls api.getUserData() inside

    ... expecting exception but getting real api call
}
```

I Also tried `declareMock<UserViewModel>{ given(this.api.getUserData()} ... `but had `NullPointerException` on `this.api`.

**TL;DR** How to properly stub already injected bean inside viewModel for UnitTesting on-the-fly
